drivers/wifi/esp_at/esp.c

```diff
@@ -925,8 +952,27 @@ static void esp_init_work(struct k_work *work)
 #endif
 		/* enable multiple socket support */
 		SETUP_CMD_NOHANDLE("AT+CIPMUX=1"),
-		/* only need ecn,ssid,rssi,channel */
+
+#if defined(CONFIG_WIFI_ESP_AT_SCAN_MAC_ADDRESS)
+		/* We need ecn,ssid,rssi,mac,channel */
+#if defined(CONFIG_WIFI_ESP_AT_SCAN_RESULT_RSSI_ORDERED)
+		SETUP_CMD_NOHANDLE("AT+CWLAPOPT=1,31"),
+#else
+		SETUP_CMD_NOHANDLE("AT+CWLAPOPT=0,31"),
+#endif
+#else
+		/* no mac: only need ecn,ssid,rssi,channel */
+#if defined(CONFIG_WIFI_ESP_AT_SCAN_RESULT_RSSI_ORDERED)
+		SETUP_CMD_NOHANDLE("AT+CWLAPOPT=1,23"),
+#else
 		SETUP_CMD_NOHANDLE("AT+CWLAPOPT=0,23"),
+#endif
```

Comment:
Let's create some helper macro like `ESP_CMD_CWLAPOPT` depending on the configuration, so that this function does not get that big.

Also, the first parameter to `AT+CWLAPOPT=`, which is `0` or `1`, seems to be dependent on `CONFIG_WIFI_ESP_AT_SCAN_RESULT_RSSI_ORDERED`, so let's create a helper for that as well, e.g.:
```
#if defined(CONFIG_WIFI_ESP_AT_SCAN_RESULT_RSSI_ORDERED)
#define ESP_CMD_CWLAPOPT_ORDERED "1"
#else
#define ESP_CMD_CWLAPOPT_ORDERED "0"
#endif
```

Reply:
Added macros